src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private int currentCapacity;
+    private int currentLoadFactor;
+    private Node<K, V>[] storage;
+
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        currentLoadFactor = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        storage = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeed();
+        int position = getPosition(key);
+        Node<K, V> oldNode = storage[position];
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (oldNode == null) {
+            storage[position] = newNode;
+            size++;
+        } else {
+            Node<K, V> nextNode = oldNode;
+            while (nextNode != null) {
+                if (Objects.equals(nextNode.key, newNode.key)) {
+                    nextNode.value = newNode.value;
+                    break;
+                }
+                if (nextNode.nextNode == null) {
+                    nextNode.nextNode = newNode;
+                    size++;
+                    break;
+                } else {
+                    nextNode = nextNode.nextNode;
```

Comment:
You may use empty return; to stop method. It simplifies if-else constructions.
Also you may switch to store reference of previous node instead of checking forward every time, it will decrease null-checks.

Reply:
ok, good idea